uvicorn/config.py

```diff
@@ -446,6 +448,9 @@ def load(self) -> None:
                 ca_certs=self.ssl_ca_certs,
                 ciphers=self.ssl_ciphers,
             )
+
+        elif self.ssl_context:
+            self.ssl = self.ssl_context.custom_ssl_context_factory()  # type: ignore
```

Comment:
```suggestion
            self.ssl = self.ssl_context() 
```
this should do it no ? I dont get it otherwise